test/puppetlabs/puppetdb/command_test.clj

```diff
@@ -1883,6 +1883,49 @@
            (testing "concurrent-depth metric should be 0"
              (is (= 0 (concurrent-depth))))))))))
 
+(deftest block-command-processing
```

Comment:
I'm not positive yet, but I suspect we might be able to remove the need for the process-message `with-redefs` here by just waiting on changes to the `(-> cmd-service-context :stats)` atom, i.e. it has `:received-commands` and `:executed-commands` counts, and we could just pause commands, submit the command, wait for `:received-commands` to go to 1, then make sure `:executed-commands` stays zero for more than a second or two, unpause commands, and then wait for `:executed-commands` to go to 1.  If we end up thinking that seems reasonable, then `await-ref-state` might be useful.